packages/core/components/DirectoryTree/useDirectoryHierarchy.tsx

```diff
@@ -128,7 +131,8 @@ const useDirectoryHierarchy = (
         async function getContent() {
             if (isLeaf || hierarchy.length === 0) {
                 // if we're at the top or bottom of the hierarchy, render a FileList
-                if (!cancel) {
+                // unless we have cancelled or there is nothing to query against
+                if (!cancel && !isEmpty(fileSet.toJSON().queryString)) {
```

Comment:
For some reason when I use a csv datasource this condition is never getting fulfilled for me unless I click new filter selections. It just shows the loading spinner forever and never dispatches `receiveContent`

Reply:
Also loads forever & doesn't notify user if it can't find the file anymore; hides error in console instead of triggering modal

Reply:
Total flop, I added this bit when my query had a grouping in it so I wasn't hitting this. I removed it in favor of checking explicitly for the data source being ready in https://github.com/AllenInstitute/biofile-finder/pull/255/commits/616489193eb8ef2f473bf0ce432b263bfab0c4f9